mail-integration-services/src/main/java/org/exoplatform/mailintegration/rest/MailIntegrationRest.java

```diff
@@ -190,4 +184,36 @@ public Response getMessageById(@ApiParam(value = "Message id", required = true)
       return Response.status(Response.Status.INTERNAL_SERVER_ERROR).build();
     }
   }
+
+  @PUT
+  @Consumes(MediaType.APPLICATION_JSON)
+  @Produces(MediaType.APPLICATION_JSON)
+  @RolesAllowed("users")
+  @ApiOperation(value = "Update an existing mail integration setting", httpMethod = "PUT", response = Response.class, consumes = "application/json")
+  @ApiResponses(
+          value = { @ApiResponse(code = HTTPStatus.NO_CONTENT, message = "Request fulfilled"),
+                  @ApiResponse(code = HTTPStatus.NOT_FOUND, message = "Object not found"),
```

Comment:
a check on mailIntegrationSetting existance is needed as for deleteMailIntegrationSetting

Reply:
it had been done